### PR TITLE
Add refresh token revocation implicit triggers

### PR DESCRIPTION
**Pull request by royalchan-okta**
OKTA-113844

@larsjohansen-okta @richardmateosian-okta 

image

### CODE REVIEW

File: _source/_docs/api/resources/oauth2.md [hunk 1]
```diff
@@ -248,16 +248,25 @@ A Refresh Token will be returned if 'offline_access' scope is requested using au
...
+
 ## ID Token
-An authorization server can also issue an Id Token to the client, just like [OIDC](oidc#id-token). The differences are 1) 'groups' is not a reserved scope or claim. To get a claim with group information, the administrators have to define a custom claim with a group filter and associate it with a scope. 2) the custom properties in the app user profile will not be put in the Id Token by default even if profile scope is granted. To get a claim for a custom property, the administrators have to define a custom claim with an Okta EL expression and associate it with a scope.
+An Authorization Server can also issue an Id Token to the client, just like [OIDC](oidc#id-token). The differences are 1) 'groups' is not a reserved scope or claim. To get a claim with group information, the administrators have to define a custom claim with a group filter and associate it with a scope. 2) the custom properties in the app user profile will not be put in the Id Token by default even if profile scope is granted. To get a claim for a custom property, the administrators have to define a custom claim with an Okta EL expression and associate it with a scope.
```
Comment (richardmateosian-okta) [inline _source/_docs/api/resources/oauth2.md]: I would reword this slightly:

An Authorization Server can issue an ID Token to the client, as in [OIDC](oidc#id-token), but with the following differences: 
 * The ID Token cannot contain a reserved scope or claim called 'groups'. To obtain a claim with group information, administrators must define a custom claim with a group filter and associate it with a scope.
 * The custom properties in the app user profile are not included in the Id Token by default, even if profile scope is granted. To obtain a claim for a custom property, administrators must define a custom claim with an Okta Expression Language expression and associate it with a scope.

File: _source/_docs/api/resources/oauth2.md [hunk 2]
```diff
@@ -248,16 +248,25 @@ A Refresh Token will be returned if 'offline_access' scope is requested using au
...
+### Refresh Token Revocation
+
+Refresh Tokens can be revoked explicitly by making a [Revocation Request](#revocation-request). Additionally, all Refresh Tokens associated with an entity will be revoked when the entity is deactivated, deleted, or otherwise modified in a way that invalidates the associated Refresh Tokens. This includes:
+
+* The User is Suspended or Deactivated
+* The Client App is Deactivated or Deleted
+* The Authorization Server's Resource URI is modified, or the Authorization Server is deleted.
+
+All other modifications do not affect existing tokens.
+
```
Comment (richardmateosian-okta) [inline _source/_docs/api/resources/oauth2.md]: I would say "No other modifications affect existing tokens."

File: _source/_docs/api/resources/oauth2.md [hunk 3]
```diff
@@ -248,16 +248,25 @@ A Refresh Token will be returned if 'offline_access' scope is requested using au
 
 The lifetime of a Refresh Token is configured in [Access Policies](#access-policies), the minimum value is 24 hours. The refresh token can also expire after a period if no clients redeem it for an Access Token. The period should be equal to or larger than 10 minutes. If the token's lifetime is set to unlimited, the Authorization Server will not check if clients use it or not.
 
+### Refresh Token Revocation
+
+Refresh Tokens can be revoked explicitly by making a [Revocation Request](#revocation-request). Additionally, all Refresh Tokens associated with an entity will be revoked when the entity is deactivated, deleted, or otherwise modified in a way that invalidates the associated Refresh Tokens. This includes:
+
```
Comment (richardmateosian-okta) [inline _source/_docs/api/resources/oauth2.md]: I would change "will be" to "are" and change "This includes" to "Such ways include"

File: _source/_docs/api/resources/oauth2.md [hunk 4]
```diff
@@ -248,16 +248,25 @@ A Refresh Token will be returned if 'offline_access' scope is requested using au
...
 ## ID Token
-An authorization server can also issue an Id Token to the client, just like [OIDC](oidc#id-token). The differences are 1) 'groups' is not a reserved scope or claim. To get a claim with group information, the administrators have to define a custom claim with a group filter and associate it with a scope. 2) the custom properties in the app user profile will not be put in the Id Token by default even if profile scope is granted. To get a claim for a custom property, the administrators have to define a custom claim with an Okta EL expression and associate it with a scope.
+An Authorization Server can also issue an Id Token to the client, just like [OIDC](oidc#id-token). The differences are 1) 'groups' is not a reserved scope or claim. To get a claim with group information, the administrators have to define a custom claim with a group filter and associate it with a scope. 2) the custom properties in the app user profile will not be put in the Id Token by default even if profile scope is granted. To get a claim for a custom property, the administrators have to define a custom claim with an Okta EL expression and associate it with a scope.
```
Comment (larsjohansen-okta) [inline _source/_docs/api/resources/oauth2.md]: I realize your only change here was the capitalization of "Authorization Server", but since you made me aware of it - let's try to be consistent with the capitalization of "ID Token" as well. There's two examples of "Id Token" in this section, but we use "ID Token" everywhere else (as far as I know).

Comment (larsjohansen-okta) [inline _source/_docs/api/resources/oauth2.md]: Ugh, just read through this page, and it seems like we use "ID token" as well somewhere, maybe we can do a grep for "id token" and make it consistent across all of our docs.
